.travis.yml

```diff
@@ -0,0 +1,105 @@
+#------------------------------------------------------------------------------
+# TravisCI configuration file for cpp-ethereum.
+#
+# The documentation for cpp-ethereum is hosted at:
+#
+# http://www.ethdocs.org/en/latest/ethereum-clients/cpp-ethereum/
+#
+# ------------------------------------------------------------------------------
+# This file is part of cpp-ethereum.
+#
+# cpp-ethereum is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cpp-ethereum is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2016 cpp-ethereum contributors.
+#------------------------------------------------------------------------------
+
+language: cpp
+branches:
+    only:
+        - develop
+        - merge_repos
+        - release
+    except:
+        - /develop-v[0-9]/
+        - /untagged/
+matrix:
+    include:
+        - os: linux
+          dist: trusty
+          sudo: required
+          env:
+              - TRAVIS_BUILD_TYPE=RelWithDebInfo
+              - ZIP_SUFFIX=ubuntu-trusty
+              - TESTS=On
+        - os: osx
+          osx_image: beta-xcode6.2
+          env:
+              - TRAVIS_BUILD_TYPE=RelWithDebInfo
+              - ZIP_SUFFIX=osx-mavericks
+              - TESTS=On
+        - os: osx
+          osx_image: xcode7.1
+          env:
+              - TRAVIS_BUILD_TYPE=RelWithDebInfo
+              - ZIP_SUFFIX=osx-yosemite
+              - TESTS=Off
+        - os: osx
+          osx_image: xcode7.3
+          env:
+              - TRAVIS_BUILD_TYPE=RelWithDebInfo
+              - ZIP_SUFFIX=osx-elcapitan
+              - TESTS=Off
+        - os: osx
+          osx_image: xcode8
+          env:
+              - TRAVIS_BUILD_TYPE=RelWithDebInfo
+              - ZIP_SUFFIX=macos-sierra
+              - TESTS=Off
+git:
+    depth: 2
+cache:
+    - ccache
+    - directories: build
+install:
+    - ./scripts/install_deps.sh
+before_script:
+    - mkdir -p build
+    - cd build
+    - cmake .. -DCMAKE_BUILD_TYPE=$TRAVIS_BUILD_TYPE -DTESTS=$TESTS
+    - make -j2
+    - ../scripts/release.sh $ZIP_SUFFIX $TESTS
```

Comment:
I'll have a look at the ordering, but this isn't actually harmful even in that case, because `release.sh` generates the ZIP but doesn't publish it until later (after tests pass).    It only takes a few seconds, and would just result in a little redundant work and an unused ZIP on the machine.